### ISSUE TITLE
Feature Request: SelectList with search

### ISSUE THREAD

**Issue by leroy0211** [jquense/react-widgets#675]
I think it would be nice to have a search field in `SelectList` when required. 
It shouldn't be enabled by default, but has to be configured. 

Any thoughts?

**Comment by jquense** [comment]
Hi there! The UI would be a bit weird there I think, You'd really want to control how and where the search bar appears so you could integrate it nicely into the surrounding app, vs the more encapsulated feel of the dropdown searches. I'll keep it in mind for the future, but no immediate plans to implement such a thing

**Comment by leroy0211** [comment]
I have created a new Component myself to coop with this. Below is what I meant by having a SelectList with a search bar for long lists.

![react_app](https://user-images.githubusercontent.com/2051658/34606874-52b8e838-f211-11e7-9f2c-7b214e48e36b.png)
![react_app](https://user-images.githubusercontent.com/2051658/34606895-625509ca-f211-11e7-9fc9-e6d0503d5521.png)

And in addition I have added another SelectList which groups the selected options. The disabled state of an item is optional, as would be the unlisted state.

Disabled state:
![react_app](https://user-images.githubusercontent.com/2051658/34607058-1e4b5454-f212-11e7-8438-4e296d5ebca1.png)

Unlisted state:
![react_app](https://user-images.githubusercontent.com/2051658/34607069-318e369e-f212-11e7-9739-b9753d9017c4.png)
